12-offer-encoding.md

```diff
@@ -302,6 +304,12 @@ A reader of an offer:
         from that estimate.
   - SHOULD not respond to an offer if the current time is after
     `offer_absolute_expiry`.
+  - if it chooses to sends an `invoice_request`, it sends an onion message:
+    - if `offer_node_id` is set:
+      - MUST send the onion message to `offer_node_id`
```

Comment:
Given the writer MAY set `offer_node_id` to the node's public key when `offer_paths` is set, the reader can't assume it *is* the node's public key -- it may be transient! So reader should only send to `offer_node_id` when `offer_paths` is _not_ set.

Reply:
This seems very wrong to me.  If you don't want them prove a offer_node_id, don't set it.  But offer_paths ids are *forced transient*: they are useless for any reputation system (they sign the invoice with that random key, but that means nothing).  

Public nodes will want to sign their invoices with their node_id, so will any vendors trying to build a reputation for honesty.  See Matt's DNS proposal, for example.  Such offers may also want to use blinded paths, as they're the only way to give capacity or routing hints.

Reply:
The reason for setting both `offer_paths` and `offer_node_id` would be to enjoy the privacy of blinded paths while still using a stable identity that user can attach a reputation to.
Example: I'm a merchant and I want to use a stable node id that user can learn to trust, so I set `offer_node_id` and sign invoices with it. However I either don't have a public node (e.g. phoenix) or don't want it to be known that it is my node, so I use `offer_paths` and set `offer_node_id` to be different from the node id of my lightning node.
I also disagree with your assertion that blinded paths are necessarily transient, all phoenix wallets will use blinded paths that will be valid forever.